channel/pallet/adjudicator.go

```diff
@@ -142,6 +147,24 @@ loop:
 	}
 }
 
+func (a *Adjudicator) waitProgressable(ctx context.Context, ch pchannel.ID) error {
+	// Fetch on-chain dispute.
+	dis, err := a.pallet.QueryStateRegister(ch, a.storage, a.pastBlocks)
+	if err != nil {
+		return err
+	}
+
+	// If we are in the register phase, we wait for the dispute timeout.
+	if dis.Phase == channel.RegisterPhase {
+		timeout := channel.MakeTimeout(dis.Timeout, a.storage)
+		err := timeout.Wait(ctx)
```

Comment:
Doesn't the timeout get restarted when a new state is registered in the meantime?

Reply:
Otherwise, if we ignore that case, LGTM.

Reply:
Only in the Force Execution phase. The registration timeout does not change once the dispute is initiated.